Automattic/newspack-blocks
# Improve Homepage Posts block editor performance

### All Submissions:






### Changes proposed in this Pull Request:

Improve the performance of Homepage Blocks post in the editor. 



Closes #493 

### How to test the changes in this Pull Request:

1. Create a page with `n` of Homepage Posts blocks, each with a category assigned
2. On master, load the page in the editor. Observe a larger* than `n` number of requests to `/v2/posts` WP API endpoint**
3. Switch to this branch and rebuild
4. Observe that the number of `/v2/posts` requests on editor page load matches the number of blocks on the page
4. Observe that some of the block posts query controls are disabled during the update***
5. Simulate an error by blocking the first `/v2/posts` request****. Observe that the sabotaged block displays an error, but the rest are displayed as expected. Such an error on `master` should  cause an infinite loading state on the block.  

\* I went with 10, which triggered 55 requests
\** To see the no. of requests in Chrome's network tab:

![image](https://user-images.githubusercontent.com/7383192/83127005-e9dacf00-a0d9-11ea-9aaa-c81ad217d800.png)

\*** Not all `@wordpress/components` control components support the `disabled` prop - changing that is in progress at https://github.com/WordPress/gutenberg/pull/22701

\**** In Chrome network tab:
![image](https://user-images.githubusercontent.com/7383192/83129557-42f83200-a0dd-11ea-9a7f-44e4062bba76.png)


### Other information:
